app/src/main/java/net/gsantner/markor/format/asciidoc/AsciidocSyntaxHighlighter.java

```diff
@@ -0,0 +1,324 @@
+    private static final int AD_BACKCOLOR_DARK_MONOSPACE = TOL_DARK_GRAY;
+    private static final int AD_BACKCOLOR_LIGHT_SQUAREBRACKETS = TOL_PALE_GRAY;
+    private static final int AD_BACKCOLOR_DARK_SQUAREBRACKETS = TOL_DARK_GRAY;
+    private static final int AD_BACKCOLOR_LIGHT_BLOCKTITLE = TOL_PALE_GRAY;
+    private static final int AD_BACKCOLOR_DARK_BLOCKTITLE = TOL_DARK_GRAY;
+
+    // use the same highlight for light and dark theme
+    private static final int AD_BACKCOLOR_LIGHT_HIGHLIGHT = Color.YELLOW;
+    private static final int AD_BACKCOLOR_DARK_HIGHLIGHT = Color.YELLOW;
+    // black on yellow
+    private static final int AD_FORECOLOR_HIGHLIGHT = Color.BLACK;
+
+    // ..._COLOR_LIGHT_... = Color for light theme
+    // TODO: too little contrast to normal font with the TOL_ colors
+//    private static final int AD_COLOR_LIGHT_COMMENT = TOL_DARK_GRAY;
+//    private static final int AD_COLOR_DARK_COMMENT = TOL_PALE_GRAY;
+//    // https://docs.oracle.com/javase/9/docs/api/javafx/scene/paint/Color.html
+//    // use Color.GRAY instead of TOL_GRAY, because Color.GRAY is the middle between white and black: #808080
+//    // alternative: TOL_PALE_GRAY vs. TOL_DARK_GRAY
+//    private static final int AD_COLOR_COMMENT = Color.GRAY; //TOL_GRAY;
+    private static final int AD_FORECOLOR_LIGHT_COMMENT = Color.GRAY;
+    private static final int AD_FORECOLOR_DARK_COMMENT = Color.GRAY;
+
+
+    private boolean _highlightLineEnding;
+    private boolean _highlightCodeChangeFont;
+    private boolean _highlightBiggerHeadings;
+    private boolean _highlightCodeBlock;
+
+
+    public AsciidocSyntaxHighlighter(AppSettings as) {
+        super(as);
+    }
+
+    @Override
+    public SyntaxHighlighterBase configure(Paint paint) {
+        _highlightLineEnding = _appSettings.isAsciidocHighlightLineEnding();
+// TODO: does not work yet
+        _highlightBiggerHeadings = _appSettings.isAsciidocBiggerHeadings();
+        _highlightCodeChangeFont = _appSettings.isHighlightCodeMonospaceFont();
+        _highlightCodeBlock = _appSettings.isHighlightCodeBlock();
+        _delay = _appSettings.getAsciidocHighlightingDelay();
+        return super.configure(paint);
+    }
+
+    @Override
+    protected void generateSpans() {
+        createTabSpans(_tabSize);
+        // TODO: understand: what happens here?
+        createUnderlineHexColorsSpans();
+        // TODO: createSmallBlueLinkSpans() - font is very small, currently general setting: 85% of common size
+        // TODO: implement some kind of AsciiDoc link later
+        // hard to read on dark theme, but this is a general question for all formats,
+        // not AsciiDoc specific
+        // also it uses private static String formatLink(String text, String link), which is
+        // adapted for Markdown
+        // maybe, needs to be adapted for AsciiDoc?
+        // but not in the current Pull Request
+        // createSmallBlueLinkSpans();
```

Comment:
Yes font size for http links is intentionally reduced - as they very often span across 3+ lines of text while being usually not that important text wise. For link details theres also the link edit button or open link button for such cases.

Please just enable it here as well, like everywhere else.

Reply:
They don't fully match AsciiDoc Syntax. And I don't wanted to touch them in this pull request, to fully match. And a partially match is not so fine.

Reply:
yeah these are generic for all formats, when you dont have something specific around you can just enable those